plugins/@grouparoo/airtable/src/lib/destination/exportRecords.ts

```diff
@@ -0,0 +1,244 @@
+    }
+    const found = getByForeignKey(key.toString());
+    if (found) {
+      found.destinationId = record.id;
+      found.result = record.fields;
+    }
+  });
+};
+
+function attachErrorsToErroneousRecords(
+  records: BatchExport[],
+  error: AirtableError
+) {
+  records.map((record) => {
+    record.error = error;
+  });
+}
+
+function buildUpdatePayload(
+  exportedProfile: BatchExport
+): RecordData<Partial<FieldSet>> {
+  if (!exportedProfile.destinationId) {
+    throw ReferenceError(
+      `Could not find destination ID in Batch Export with key ${exportedProfile.recordId}`
+    );
+  }
+  return {
+    id: exportedProfile.destinationId,
+    fields: exportedProfile.newRecordProperties,
+  };
+}
+
+function buildCreatePayload(
+  exportedProfile: BatchExport
+): CreateRecord<FieldSet> {
+  return {
+    fields: exportedProfile.newRecordProperties,
+  };
+}
+
+const addToGroups: BatchMethodAddToGroups = async () => {
+  // This destination doesn't do groups
+};
+
+const removeFromGroups: BatchMethodRemoveFromGroups = async () => {
+  // This destination doesn't do groups
+};
+// mess with the keys (lowercase emails, for example)
+const normalizeForeignKeyValue: BatchMethodNormalizeForeignKeyValue = ({
+  keyValue,
+}) => {
+  if (!keyValue) {
+    return null;
+  }
+  return keyValue.toString().toLowerCase().trim();
+};
+
+// mess with the names of groups (tags with no spaces, for example)
+const normalizeGroupName: BatchMethodNormalizeGroupName = ({ groupName }) => {
+  return groupName.toString().trim();
+};
+interface ExportBatchOptions {
+  appOptions: SimpleAppOptions;
+  destinationOptions: SimpleDestinationOptions;
+  syncOperations?: DestinationSyncOperations;
+  exports: BatchExport[];
+}
+export async function exportBatch(exportBatchOptions: ExportBatchOptions) {
+  const { syncOperations, appOptions, destinationOptions, exports } =
+    exportBatchOptions;
+  const { primaryKey } = destinationOptions;
+  const batchSize = 10;
+  const findSize = 100;
+
+  return exportRecordsInBatch(
+    exports,
+    {
+      findSize,
+      batchSize,
+      groupMode: BatchGroupMode.TotalMembers,
+      syncMode: BatchSyncMode.Additive,
+      syncOperations,
+      appOptions,
+      destinationOptions,
+      foreignKey: primaryKey.toString(),
+    },
+    {
+      getClient,
+      findAndSetDestinationIds,
+      deleteByDestinationIds,
+      updateByDestinationIds,
+      createByForeignKeyAndSetDestinationIds,
+      addToGroups,
+      removeFromGroups,
+      normalizeForeignKeyValue,
+      normalizeGroupName,
+    }
+  );
+}
+export const exportRecords: ExportRecordsPluginMethod = async ({
+  appOptions,
+  destinationOptions,
+  syncOperations,
+  exports: recordsToExport,
+}) => {
+  const batchExports = buildBatchExports(recordsToExport);
+  try {
+    return await exportBatch({
+      appOptions,
+      destinationOptions,
+      syncOperations,
+      exports: batchExports,
+    });
+  } catch (error) {
```

Comment:
probably not needed here. though this is the place where we'd do `429` rate-limiting kinds of things. ever see anything like that?

Reply:
I was thinking about doing that but this is currently done in the Airtable client supposedly internally and the actions not handled by their client would only be done with human involvement during initial configuration. 
[Link to exponential backoff implementation in Airtable JS repository](https://github.com/Airtable/airtable.js/blob/9d40666979af77de9546d43177cab086a03028bf/src/run_action.ts#L67)

Reply:
hmm, we'd want to handle that probably. otherwise, they'd tie up the thread.
Tjhis is fine for now, but we can investigate setting `_noRetryIfRateLimited` later.